### PR TITLE
CAMEL-12554 - camel-geocoder - Use new API

### PR DESCRIPTION
**Pull request by jborza**
Hi,

as described in 
https://issues.apache.org/jira/browse/CAMEL-12554

Geocoding and reverse geo-ip is now using a single library: Google Maps API Web Services

Not sure if we usually remove component options in Camel, but there have been a few changes as follow:

- I had to remove the options httpClientConfigurer, httpConnectionManager as the http connections are no longer managed by this library.
- I also removed proxyAuthDomain, proxyAuthHost and proxyAuthMethod=NTLM as the new library doesn't support that either.

Also as the API requires an API key the tests attempt to retrieve it from CAMEL_GEOCODER_APIKEY. Again, not sure what is the best practice - probably the same thing we'd do in Azure/AWS integration which probably requires some kind of API key to be included as well.

Other option may be using a mock Google Maps endpoint, which I'm a bit reluctant to write as it may change any time.

Regards,
Juraj

### CODE REVIEW

**Comment by oscerd** [comment]
If we are changing this much, it may have sense to create a completely new component, deprecating camel-geocoder and suggest to use the new one, otherwise users may be confused when upgrading. @davsclaus thoughts?

**Comment by davsclaus** [comment]
Thanks for this PR - that is awesome.

@oscerd the public API of the old api is being terminated this month, so it does not really matter if there is some migration effort etc. Anyone would need to migrate/upgrade to keep using it.

**Comment by oscerd** [comment]
Ok, lets merge this then

File: components/camel-geocoder/src/main/java/org/apache/camel/component/geocoder/GeoCoderEndpoint.java [hunk 1]
```diff
@@ -158,6 +149,15 @@ public void setClientKey(String clientKey) {
         this.clientKey = clientKey;
     }
 
+
+    private String getApiKey() {
+        return apiKey;
+    }
+
+    public void setApiKey(String apiKey) {
```
Comment (onderson) [inline components/camel-geocoder/src/main/java/org/apache/camel/component/geocoder/GeoCoderEndpoint.java]: missing javadoc and component validation fails.

File: components/camel-geocoder/src/test/java/org/apache/camel/component/geocoder/GeoCoderComponentAddressTest.java [hunk 2]
```diff
@@ -39,10 +39,10 @@ protected RouteBuilder createRouteBuilder() throws Exception {
         return new RouteBuilder() {
             public void configure() {
                 from("direct:start")
-                    .to("geocoder:address:Paris, France")
-                    .to("log:result")
-                    .log("Location ${header.CamelGeocoderAddress} is at lat/lng: ${header.CamelGeocoderLatlng} in city ${header.CamelGeocoderCity}")
-                    .to("mock:result");
+                        .to("geocoder:address: empty?apiKey=" + getApiKey())
```
Comment (onderson) [inline components/camel-geocoder/src/test/java/org/apache/camel/component/geocoder/GeoCoderComponentAddressTest.java]: tests do no run. not sure about this uri setting.

**Comment by onderson** [comment]
#2416 closes #2415

**Comment by onderson** [comment]
thanks @jborza , did minor changes on your PR and merged and closed this one.

File: components/camel-geocoder/src/test/java/org/apache/camel/component/geocoder/GeoCoderComponentAddressTest.java (hunk 2 above)
Comment (jborza) [reply components/camel-geocoder/src/test/java/org/apache/camel/component/geocoder/GeoCoderComponentAddressTest.java]: @onderson are there other tests that require secrets?

Comment (onderson) [reply components/camel-geocoder/src/test/java/org/apache/camel/component/geocoder/GeoCoderComponentAddressTest.java]: @jborza , not sure, as i said no tests were run while doing the build for the component.

**Pull request closed without merging**